userstorage/backend.py

```diff
@@ -94,6 +96,8 @@ def __enter__(self):
         """
         Setup the current storage and return it.
         """
+        if not self.exists() and not self.required:
```

Comment:
Why check if the storage is required? maybe add a comment explaining the check?

Reply:
Inferred it from this comment: https://github.com/nirs/userstorage/pull/11#discussion_r969593132

But now that I recheck with the code, I think you meant it as in required storage will fail during `create`, and non-required storage will xfail during `setup`, so I will remove `required` from this condition.